packages/shared/src/components/widgets/SocialShare.tsx

```diff
@@ -106,6 +94,14 @@ export const SocialShare = ({
           emailTitle="I found this amazing post"
         />
       </SocialShareContainer>
+      {squadToShare && (
+        <CreateSharedPostModal
```

Comment:
There is a dep-cycle issue that arises if we use `LazyModal` for this. imo, it is better to have a single instance of not using the LazyModal flow rather than numerous instances of `ShareModal`.

Reply:
Maybe add a comment into the code as well, so that we know why it's done like this? :)

Reply:
Makes sense. Let me add that.